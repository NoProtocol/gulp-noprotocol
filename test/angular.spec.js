const gulp = require("gulp");
const streamAssert = require("stream-assert-gulp");
const noprocotol = require("../");
const fixtureDir = __dirname + "/fixtures/";

describe("noprocotol.angular", function() {
  it("generates a js bundle and a sourcemap file", function(done) {
    gulp
      .src(fixtureDir + "controller.js")
      .pipe(
        noprocotol.angular({
          deps: ["ngAnimate"] // Genereate an angular.module()
        })
      )
      .pipe(streamAssert.length(2))
      .pipe(
        streamAssert.first(function(file) {
          expect(file.relative).toBe("app.min.js.map");
        })
      )
      .pipe(
        streamAssert.second(function(file) {
          expect(file.relative).toBe("app.min.js");

          expect(file.contents.toString()).toMatchSnapshot();
        })
      )
      .on("end", done);
  });

  it("places html files into the $templateCache", function(done) {
    gulp
      .src(fixtureDir + "**/hello.html")
      .pipe(noprocotol.angular())
      .pipe(streamAssert.length(2))
      .pipe(
        streamAssert.second(function(file) {
          expect(file.relative).toBe("app.min.js");
          expect(file.contents.toString()).toMatchSnapshot();
        })
      )
      .on("end", done);
  });
});
