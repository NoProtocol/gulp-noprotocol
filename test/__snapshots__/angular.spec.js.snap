// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`noprocotol.angular generates a js bundle and a sourcemap file 1`] = `
"\\"use strict\\";var app=angular.module(\\"app\\",[\\"ngAnimate\\"]);app.controller(\\"TestCtrl\\",[\\"$scope\\",function(r){r.x;return!0}]);
//# sourceMappingURL=app.min.js.map
"
`;

exports[`noprocotol.angular places html files into the $templateCache 1`] = `
"\\"use strict\\";!function(){var t;try{t=angular.module(\\"app\\")}catch(l){t=angular.module(\\"app\\",[])}t.run([\\"$templateCache\\",function(t){t.put(\\"/test/fixtures/hello.html\\",\\"<h1>Hello</h1><p>world</p>\\")}])}();
//# sourceMappingURL=app.min.js.map
"
`;
